iree/compiler/Conversion/LinalgToNVVM/test/convert_to_nvvm.mlir

```diff
@@ -0,0 +1,27 @@
+// RUN: iree-opt -iree-codegen-convert-to-nvvm %s | IreeFileCheck %s
+
+func @abs_ex_dispatch_0() {
+  %0 = iree.placeholder for "interface buffer" {binding = @legacy_io::@ret0, operand_result_index = 2 : i32} : memref<16xf32>
+  %1 = iree.placeholder for "interface buffer" {binding = @legacy_io::@arg0, operand_result_index = 0 : i32} : memref<16xf32>
+  %2 = iree.placeholder for "interface buffer" {binding = @legacy_io::@arg1, operand_result_index = 1 : i32} : memref<16xf32>
```

Comment:
iree.placeholder is deprecated and going to be removed asap (spirv is the only thing using it, and I think/hope its use may be going away with the linalg on tensors work) - it's fundamentally broken and one of the things blocking buffer allocation as it doesn't support aliasing/offsetting/etc as required when bindings are not 1:1 with the I/O buffers. You should be using `hal.interface.binding.subspan`; see: https://github.com/google/iree/blob/c9cc42e313ec90d4cb2134f4a12aad87d6eaf848/iree/compiler/Conversion/LinalgToLLVM/test/hal_interface_bindings.mlir#L32-L59

I'm hesitant to add more code that depends on this deprecated path - I think we should be building out NVVM on the linalg on tensors path exclusively as otherwise we are piling on more tech debt.

/cc @MaheshRavishankar am I right? about iree.placeholder being able to go away when SPIRV is using linalg on tensors?

Reply:
Thanks for pointing that out, I didn't know. I agree it doesn't make sense to add it in NVVM path if it is deprecated. I had originally tried to match LLVM backend, I see now that there is a comment about it being deprecated.
I need to use linalg on tensor path to be able to avoid getting `iree.placeholder`?

Reply:
Trying it out, even on the CPU pass I see that by default we go through linalg on buffer path and generate `iree.placeholder` when trying to enable `-iree-codegen-llvm-experimental-linalg-on-tensors` on a simpl hlo.add it crashes as it seems like some bufferization is missing? I see a bunch of `hal.interface.load.tensor` making convertToLLVM fail.
@MaheshRavishankar , @benvanik do you know when we think we'll have a stable without using `HloToLinalgOnBuffer` pass? If it is soon I can wait for it and match what the CPU does otherwise I would prefer pushing the code with the LinalgOnBuffer flow even if it gets broken later since there is a lot of boiler plate that would be convenient to land.

Reply:
(I left a comment, dont know where it went), 

You need to add two flags to enable the linalg on tensors path `-iree-flow-dispatch-linalg-on-tensors` and the one you mentioned above. (`-iree-codegen-spirv-experimental-linalg-on-tensors` for the SPIR-V path)

If you see here https://github.com/google/iree/blob/main/iree/test/e2e/xla_ops/BUILD AFAIK, the linalg on tensors path is enabled for many of the ops (definitely add) for both LLVM and SPIR-V paths. So not sure why you are getting a failure.

The `HLOToLinalgOnBufferPass` should be deprecated when we move to Linalg on tensors

Reply:
Thanks Mahesh, I changed the code to switch to use linalg on tensor only.